.github/workflows/main.yml

```diff
@@ -0,0 +1,60 @@
+name: Main
+
+on:
+  push:
+    branches: [ main, develop, wip ]
+  pull_request:
+    branches: [ main, develop ]
+  schedule:
+    # Every twelve (12) hours.
+    # @see https://crontab.guru/every-12-hours
+    - cron: 0 */12 * * *
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: "${{ matrix.os }}-latest"
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu ]
+        php: [ 8.3 ]
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4 # https://github.com/marketplace/actions/checkout
+
+      - name: Set up PHP CLI
+        uses: shivammathur/setup-php@v2 # https://github.com/marketplace/actions/setup-php-action
+        with:
+          php-version: "${{ matrix.php }}"
+
+      - name: Create project
+        run: |
+          sudo mkdir -p /var/www
+          sudo rm -rf /var/www/*
+          sudo chmod 777 /var/www
+          composer config --global repositories.drupal-suite vcs $(pwd)
+          composer create-project --stability=dev drupal/recommended-project /var/www
+
+      - name: Set up Apache on Linux
+        if: runner.os == 'Linux'
+        run: |
+          # Install Apache w/ PHP.
+          sudo add-apt-repository ppa:ondrej/php
+          sudo apt update
+          sudo apt install \
+            apache2 \
+            php${{ matrix.php }} \
+            libapache2-mod-php${{ matrix.php }}
+          sudo a2enmod php${{ matrix.php }}
+
+          # Change the default site web directory.
+          sudo sed -i 's/\/var\/www\/html/\/var\/www\/web/g' /etc/apache2/sites-enabled/000-default.conf
+
+          # Start the service.
+          sudo service apache2 start
```

Comment:
This definitely works, but I'm wondering if there's any possibility of using Apache as a pre-existing Docker service, rather than setting it up ourselves here. Pretty sure GitHub Actions supports that...?

Reply:
> Although it might mean we can never test with Windows or macOS, since those don't have Docker support IIRC.

That agrees with my understanding, but I could look deeper.